src/pages/ocpDetails/priceListModal.tsx

```diff
@@ -0,0 +1,94 @@
+import { Modal } from '@patternfly/react-core';
+import { css } from '@patternfly/react-styles';
+import { ProviderType } from 'api/providers';
+import { AxiosError } from 'axios';
+import { ErrorState } from 'components/state/errorState/errorState';
+import { LoadingState } from 'components/state/loadingState/loadingState';
+import React from 'react';
+import { InjectedTranslateProps, translate } from 'react-i18next';
+import { connect } from 'react-redux';
+import { createMapStateToProps, FetchStatus } from 'store/common';
+import { priceListActions, priceListSelectors } from 'store/priceList';
+import { providersSelectors } from 'store/providers';
+import { modalOverride, styles } from './historicalModal.styles';
+import PriceListTable from './priceListTable';
+
+interface Props extends InjectedTranslateProps {
+  isOpen: boolean;
+  close: typeof priceListActions.closeModal;
+  fetch: typeof priceListActions.fetchPriceList;
+  providers: any;
+  name: string;
+  priceList: any;
+  priceListError: AxiosError;
+  priceListStatus: FetchStatus;
+}
+
+class PriceListModalBase extends React.Component<Props> {
+  public componentDidMount() {
+    this.props.fetch();
```

Comment:
I see that you moved the fetch to the price list modal. Unfortunately, this still performs an unnecessary request. Simply including this modal anywhere in the page would generate a request because it's based on componentDidMount. 

<img width="613" alt="Screen Shot 2019-04-12 at 12 06 20 PM" src="https://user-images.githubusercontent.com/17481322/56051022-69adaa80-5d1b-11e9-9d83-5cf09e3f613b.png">

Ideally, the fetch would be moved to PriceTable. That way, the fetch is only performed when the user launches the modal.

If you look at HistoricalModal, you'll see that the request is generated when the modal is opened. That is, HistoricalChart is lazily loaded.

FYI, we had to do this because there were too many unused requests in the page, slowing performance unnecessarily.